packages/gatsby-image/src/index.js

```diff
@@ -82,6 +82,10 @@ const hasNativeLazyLoadSupport =
   typeof HTMLImageElement !== `undefined` &&
   `loading` in HTMLImageElement.prototype
 
+// Save-Data support: https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/save-data/#detecting_the_save-data_setting
+const hasSaveDataEnabled = () =>
+  `connection` in navigator && navigator.connection.saveData === true
```

Comment:
`navigator` global object doesn't exist when generating html pages (we are running in Node.js and not in browser). So we will need to use something like `` typeof window === `undefined` `` to not crash during html generation

Question is: what do we want to display when JavaScript is disabled or in first mount (hydration doesn't patch differences in attributes - see notes in https://reactjs.org/docs/react-dom.html#hydrate). Should we use what we currently use (so include srcSet) or not?

Reply:
oh, we do have `isBrowser` thingie just below this - so we can incorporate that in our check